code/game/objects/items/weapons/stock_parts.dm

```diff
@@ -311,6 +311,53 @@
 	rating = 6
 	matter = list(MATERIAL_PLASTIC = 3, MATERIAL_GLASS = 1)
 	spawn_blacklisted = TRUE
+	
+// debug stock parts - rating 100, intended for debugging
+
+/obj/item/weapon/stock_parts/capacitor/debug
+	name = "bluespace capacitor"
+	desc = "An ultra-advanced battery able to store immense amounts of energy in a localized bluespace pocket. Used in construction of certain devices."
+	icon_state = "debug_capacitor"
+	origin_tech = list(TECH_POWER = 6, TECH_MATERIAL = 6)
+	rating = 100 // rating doesn't really matter past a certain point - this makes autolathes print stuff at 1/5th the normal cost (item that costs 5 steel now costs 1 steel)
+	matter = list(MATERIAL_STEEL = 2, MATERIAL_PLASTIC = 1, MATERIAL_GLASS = 3, MATERIAL_PLASMA = 1)
+	spawn_blacklisted = TRUE
+
+/obj/item/weapon/stock_parts/scanning_module/debug
+	name = "bluespace scanning module"
+	desc = "An ultra-advanced electronic that can rapidly scan objects, as well as their mirror images in bluespace in order to easily understand every detail. Used in construction of certain devices."
+	icon_state = "debug_scan_module"
+	origin_tech = list(TECH_MAGNET = 6)
+	rating = 100
+	matter = list(MATERIAL_STEEL = 2, MATERIAL_PLASTIC = 2, MATERIAL_GLASS = 1, MATERIAL_PLASMA = 1)
+	spawn_blacklisted = TRUE
+
+/obj/item/weapon/stock_parts/manipulator/debug
+	name = "bluespace yocto-manipulator"
+	desc = "An ultra-advanced device that can manipulate an objects molecular structure. Used in construction of certain devices."
+	icon_state = "debug_mani"
+	origin_tech = list(TECH_MATERIAL = 6, TECH_DATA = 3)
+	rating = 100
+	matter = list(MATERIAL_STEEL = 2, MATERIAL_PLASTIC = 2, MATERIAL_PLASMA = 1)
+	spawn_blacklisted = TRUE
+
+/obj/item/weapon/stock_parts/micro_laser/debug
+	name = "bluespace yocto-laser"
+	icon_state = "debug_micro_laser"
+	desc = "An ultra-advanced device that emits a laser that can emit any kind of laser on the spectrum, and several lasers off the spectrum. Used in construction of certain devices."
+	origin_tech = list(TECH_MAGNET = 6)
+	rating = 100
+	matter = list(MATERIAL_STEEL = 2, MATERIAL_PLASTIC = 2, MATERIAL_GLASS = 1, MATERIAL_PLASMA = 1)
+	spawn_blacklisted = TRUE
+
+/obj/item/weapon/stock_parts/matter_bin/debug
+	name = "bluespace matter bin"
+	desc = "An ultra-advanced container that opens into a localized pocket dimension meant for holding compressed matter. Used in construction of certain devices."
+	icon_state = "debug_matter_bin"
+	origin_tech = list(TECH_MATERIAL = 6)
+	rating = 100
+	matter = list(MATERIAL_PLASTIC = 3, MATERIAL_GLASS = 1, MATERIAL_PLASMA = 1)
+	spawn_blacklisted = TRUE
```

Comment:
I'd rather you use bad_type or spawn_frequency = 0

Reply:
hokay